app/search.py

```diff
@@ -494,6 +497,65 @@ def _query_fields(self, fields):  # pylint: disable=too-many-locals
                 f'{fields} "{preNonDigitAndPreDigit}"~5',
                 f'{fields} "{infNonDigitAndPreDigit}"'
             ]
+
+        # to improve the fuzzyness of the search result the search text has to be changed
+        # digit/text combinations are replaced with (digit|digit/text) p.e. 4a -> (4|4a)
+        # quorum operatar is used for query text fuzzy matching
+        if fuzzySearch:
+            # minimal length of single words that will be trimmed if only one word is in the query
+            # string
+            minLength = 5
+
+            def convert_if_digit(text):
+                if isdigit(text):
+                    digit = re.findall(r'^\d+', text)[0]
+                    if digit != text:
+                        return f'({str(digit)}|{text})'
+                return text
+
+            def generate_prefixes(input_list, min_length=5):
+
+                def max_text_length(input_list):
+                    """
+                    Calculates the maximum length of text elements in a given list.
+
+                    Args:
+                        input_list: A list of strings.
+
+                    Returns:
+                        The maximum length of text elements, or 0 if no text elements are found.
+                    """
+                    max_length = 0
+                    for element in input_list:
+                        if not isdigit(element):
+                            max_length = max(max_length, len(element))
+                    return max_length
+
+                result = []
+
+                if max_text_length(input_list) == 0:
+                    for text in input_list:
+                        result.append(convert_if_digit(text))
+                    return ' '.join(result)
+
+                while max_text_length(input_list) > min_length:
+                    for i, text in enumerate(input_list):
+                        if not isdigit(text) and len(text) >= min_length:
+                            input_list[i] = text[:-1]
+
+                    logger.debug('DEBUG %s %s', input_list, max_text_length(input_list))
+                    result.append(f"({' '.join([convert_if_digit(w) for w in input_list])})")
+
+                return f"({' | '.join(result)})"
```

Comment:
Ok I executed the code locally and see what it does, it creates permutations with various word lengths from the strings if they're longer than `min_length`. Maybe you could document that a bit?

Reply:
dibe